server/src/main/java/io/unitycatalog/server/persist/utils/FileUtils.java

```diff
@@ -1,178 +1,62 @@
-      return Paths.get(uri).toUri();
+  public static String createTableDirectory(String tableId) {
+    String directoryUriString = toStandardizedURIString(getStorageRoot() + "/tables/" + tableId);
+    URI directoryUri = URI.create(directoryUriString);
+    validateURI(directoryUri);
+    FileIO fileIO = fileIOFactory.getFileIO(directoryUri);
+    if (fileExists(fileIO, directoryUri)) {
+      throw new BaseException(ErrorCode.ALREADY_EXISTS, "Table directory already exists: " + directoryUri);
     }
+    return directoryUriString;
   }
 
-  private static URI createLocalDirectory(Path dirPath) {
-    // Check if directory already exists
-    if (Files.exists(dirPath)) {
-      throw new BaseException(ErrorCode.ALREADY_EXISTS, "Directory already exists: " + dirPath);
-    }
-    // Create the directory
+  public static boolean fileExists(FileIO fileIO, URI fileUri) {
     try {
-      Files.createDirectories(dirPath);
-      LOGGER.debug("Directory created successfully: {}", dirPath);
+      InputFile inputFile = fileIO.newInputFile(fileUri.getPath());
+      return inputFile.exists(); // Returns true if the file exists, false otherwise
     } catch (Exception e) {
-      throw new BaseException(ErrorCode.INTERNAL, "Failed to create directory: " + dirPath, e);
+      return false;
     }
-    return dirPath.toUri();
   }
 
   public static void deleteDirectory(String path) {
-    URI directoryUri = createURI(path);
+    URI directoryUri = URI.create(toStandardizedURIString(path));
     validateURI(directoryUri);
-    if (directoryUri.getScheme() == null || directoryUri.getScheme().equals("file")) {
-      try {
-        deleteLocalDirectory(Paths.get(directoryUri));
-      } catch (RuntimeException | IOException e) {
-        throw new BaseException(ErrorCode.INTERNAL, "Failed to delete directory: " + path, e);
-      }
-    } else if (directoryUri.getScheme().equals("s3")) {
-      modifyS3Directory(directoryUri, false);
-    } else {
-      throw new BaseException(
-          ErrorCode.INVALID_ARGUMENT, "Unsupported URI scheme: " + directoryUri.getScheme());
-    }
+    FileIO fileIO = fileIOFactory.getFileIO(directoryUri);
+    fileIO.deleteFile(directoryUri.getPath());
+    LOGGER.info("Directory deleted: " + directoryUri);
   }
 
-  private static void deleteLocalDirectory(Path dirPath) throws IOException {
-    if (Files.exists(dirPath)) {
-      try (Stream<Path> walk = Files.walk(dirPath, FileVisitOption.FOLLOW_LINKS)) {
-        walk.sorted(Comparator.reverseOrder())
-            .forEach(
-                path -> {
-                  try {
-                    Files.delete(path);
-                  } catch (IOException e) {
-                    throw new RuntimeException("Failed to delete " + path, e);
-                  }
-                });
-      }
-    } else {
-      throw new IOException("Directory does not exist: " + dirPath);
-    }
-  }
-
-  private static URI modifyS3Directory(URI parsedUri, boolean createOrDelete) {
```

Comment:
Did we move this function to some other file? Unable to find it in the PR

Reply:
This class used to only support local and s3 file systems by having code specific to these FSs. I've made changes to instead use the iceberg FileIO that the server was already using for credential vending.